amazon/ion/ioncmodule.c

```diff
@@ -372,6 +372,34 @@ static iERR ionc_write_sequence(hWRITER writer, PyObject* sequence, PyObject* tu
     Py_DECREF(sequence);
     cRETURN;
 }
+/*
+ * Process and write the key-value pair.
+ *  *  Args:
+ *      writer:  An ion writer
+ *      key: The key of IonStruct item
+ *      val: The value of IonStruct item
+ *      tuple_as_sexp: Decides if a tuple is treated as sexp
+ */
+
+static iERR write_key_value(hWRITER writer, PyObject* key, PyObject* val, PyObject* tuple_as_sexp) {
+    iERR err;
+    if (PyUnicode_Check(key)) {
+        ION_STRING field_name;
+        ion_string_from_py(key, &field_name);
+        IONCHECK(ion_writer_write_field_name(writer, &field_name));
+    } else if (key == Py_None) {
+        IONCHECK(_ion_writer_write_field_sid_helper(writer, 0));
+    }
+    IONCHECK(Py_EnterRecursiveCall(" while writing an Ion struct"));
+    err = ionc_write_value(writer, val, tuple_as_sexp);
+    Py_LeaveRecursiveCall();
+    IONCHECK(err);
+
+    return IERR_OK;
```

Comment:
If you're using `IONCHECK` I think there is an `iReturn` or some other macro that would be better. Don't have code open so IDK.

Reply:
> If you're using `IONCHECK` I think there is an `iReturn` or some other macro that would be better. Don't have code open so IDK.

Yeah that's a good catch. Here is how we define iRETURN:
` #define iRETURN       fail: RETURN(__location_name__, __line__, __count__++, err)` .
 In this case we can replace `return IERR_OK; `with `iRETURN`.